### PR TITLE
feat: generate config page

### LINKED ISSUE #11: Create generate config page
Generate

### PR DESCRIPTION
**Pull request by raosan**
### What does this PR solve?

Create What Do You Want to monitor page. Resolves #11 

### How do you implement this?

Screen Shot 2021-04-13 at 16 02 05

### How do I test this?

1. Run npm run dev
2. Open localhost:3000/advanced in browser to go to advanced page
3. click generate config file button
4. modal will be visible
5. click copy to clipboard button to copy json text

### CODE REVIEW

File: components/modal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import { Button } from '../';
+
+type ModalProps = {
+  visible: boolean;
+  onClose: () => void;
+};
+
+export default function Modal({ visible, onClose }: ModalProps): JSX.Element {
```
Comment (kevinhermawan) [inline components/modal/index.tsx:8]: I think this `Modal` component is specified to certain use case, please change the name corresponding to its functions, eg: `DownloadModal`

Comment (raosan) [reply components/modal/index.tsx:8]: You are right. will fix

Comment (raosan) [reply components/modal/index.tsx]: fixed. please check

**Review by kevinhermawan — APPROVED** [review]

**Comment by nicnocquee** [comment]
The copy to clipboard doesn't seem to work. I used it on Chrome.

But after thinking about it, maybe let's replace the "Copy To Clipboard" with "Download file" instead. So when the button is clicked, the config.json file will be downloaded.

**Comment by raosan** [comment]
> The copy to clipboard doesn't seem to work. I used it on Chrome.
> 
> But after thinking about it, maybe let's replace the "Copy To Clipboard" with "Download file" instead. So when the button is clicked, the config.json file will be downloaded.

fixed to download file

**Review by nicnocquee — APPROVED** [review]

**Pull request merged**